Library/Homebrew/rubocops/lines.rb

```diff
@@ -252,6 +252,55 @@ def autocorrect(node)
         end
       end
 
+      class ShellCommands < FormulaCop
+        def audit_formula(_node, _class_node, _parent_class_node, body_node)
+          # Match shell commands separated by spaces in the same string
+          shell_cmd_with_spaces_regex = /[^"' ]*(?:\s[^"' ]*)+/
+
+          popen_commands = [
+            :popen_read,
+            :safe_popen_read,
+            :popen_write,
+            :safe_popen_write,
+          ]
+
+          shell_metacharacters = %w[> >> < << | ; && || *]
```

Comment:
Let me know if there are additional characters that go here as well.

Reply:
Edit: I added `&`

Reply:
If this is the list of characters that mandate passing the entire command string as-is to bash, there are a ton more of interest:
* Subshell execution/substitution: `(`, `)`, `$(`, `<(`, `` ` ``
* List execution: `{`, `}`
* Arithmetic evaluation: `((`, `$((`, `))`
* Bash-style conditional expressions: `[[`, `]]`
* Variable expansions: `$`, `${`
* Glob metacharacters: `?`, `[`, `]`, `[:`, `:]`, `**`
* Extended glob metacharacters (just in case): `?(`, `*(`, `+(`, `@(`, `!(`
* Advanced redirections: `<&`, `>&`, `&>`, `&>>`, `<<<`, `<>`
* Tilde expansion: `~`

You can probably ignore the closing half of each bracket structure (i.e. `}`, `)`, etc.). I include them just for completeness.

I've left out operators that don't make sense outside the context of the above.

Reply:
Thanks!

Reply:
Current line:

```ruby
shell_metacharacters = %w[> < < | ; : & * $ ? : ~ + @ !` ( ) [ ]]
```

If characters like `:`, `+`, `@`, `!` only occur next to `(` or `[`, then I can probably remove them from the list.

Also, I didn't include `{` because it interfered with string interpolation. Any ideas for how to deal with that? I could use a regex to match `{` but not `#{`. The reason I didn't use one was that there was so much escaping because of the regex special characters.

Reply:
> Any ideas for how to deal with that?

Just omit them. This doesn't need to be 100% comprehensive.